tests/testutils/goroutineassistant.go

```diff
@@ -18,23 +18,25 @@ import (
 	"fmt"
 	"sync"
 	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
 )
 
 type GoroutineAssistant struct {
-	s  chan string
+	s  chan error
 	wg sync.WaitGroup
 	t  *testing.T
 }
 
 func NewGoroutineAssistant(t *testing.T) *GoroutineAssistant {
 	return &GoroutineAssistant{
-		s: make(chan string),
+		s: make(chan error),
 		t: t,
 	}
 }
 
 func (a *GoroutineAssistant) Fatalf(s string, args ...interface{}) {
-	a.s <- fmt.Sprintf(s, args...)
+	a.s <- fmt.Errorf(s, args...)
```

Comment:
Thanks for the reminder. I added the call to `runtime.Goexit()`. I'm not convinced to make the channel buffered, that would make the code more complex to allow dynamic channel size